src/main/java/com/google/sps/servlets/NotificationServlet.java

```diff
@@ -0,0 +1,194 @@
+  }
+
+  /**
+   * Posts a new notification for all followers of a modified question or answer.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Define local time for the new entries in the server.
+    Timestamp localTimestamp = Timestamp.valueOf(LocalDateTime.now());
+    // Check if notification is about a question answered or answer commented, along with its id.
+    String typeOfNotification = request.getParameter("type");
+    int modifiedElementId = Integer.parseInt(request.getParameter("modifiedElementId"));
+
+    // URL to which the user will be redirected.
+    String notificationUrl = "";
+    String notificationMessage = "";
+    String query = "";
+    if (typeOfNotification.equals("question")) {
+      // If the notification is for an anwer to a question.
+      query =  "SELECT follower_id FROM QuestionFollower WHERE question_id = " + modifiedElementId;
+      notificationUrl = "/question.html?id=" + modifiedElementId;
+      notificationMessage = "Your question was answered.";
+    } else if (typeOfNotification.equals("answer")) {
+      // If the notification is for a new comment in an answer.
+      query =  "SELECT follower_id FROM AnswerFollower WHERE answer_id = " + modifiedElementId;
+      notificationUrl = "/question.html?id=" + modifiedElementId;
+      notificationMessage = "Your answer was commented.";
+    }
+    // Creates notification and relationship between its ID and the ID of the concerned users.
+    createNotification(query, notificationUrl, notificationMessage, localTimestamp);
+  }
+
+  /**
+   * Fetches notifications with the user ID.
+   */
+  private List<Notification> getNotifications(int userId) {
+    // Prepare query to select notifications by the subquery of notifications IDs selected by 
+    // user ID.
+    String query =  "SELECT message, url, date_time FROM Notification WHERE id IN " +
+                    "(SELECT notification_id FROM UserNotification WHERE user_id = " + userId +
+                    ") ORDER BY date_time DESC";
+    List<Notification> notifications = new ArrayList<>();
+    // Query the information from tables and create notification object to be stored in ArrayList.
+    try (Connection connection = DriverManager.getConnection(Utility.SQL_LOCAL_URL,
+                                                             Utility.SQL_LOCAL_USER,
+                                                             Utility.SQL_LOCAL_PASSWORD);
+         PreparedStatement pst = connection.prepareStatement(query);
+         ResultSet rs = pst.executeQuery()) {
+      // Iterate through the result of the query to populate the ArrayList and return it as JSON.
+      while (rs.next()){
+        Notification notification = new Notification();
+        notification.message = rs.getString(1);
+        notification.url = rs.getString(2);
+        notification.timestamp = rs.getTimestamp(3);
```

Comment:
Use constants here rather than integers. I think it was Omar's PR that I mentioned this on, but if you use a constant like 'rs.getString(SqlConstants.NOTIFICATION_MESSAGE_INDEX)' it will be 1) more readable, and 2) harder to make mistakes and introduce bugs.

Reply:
Done!